src/bin/add/args.rs

```diff
@@ -227,7 +227,7 @@ impl Args {
                 assert!(self.path.is_none());
                 assert!(self.registry.is_none());
                 let features = get_manifest_from_url(repo)?
-                    .map(|m| m.features())
+                    .map(|m| m.features().unwrap())
```

Comment:
Ideally we wouldn't unwrap.

What if we did:
```rust
.map(|m| m.features())
.transpose()?
```
We'll have a `Option<Result<_, >>` and transpose swaps the Option and Result so we can use `?`

Reply:
That's actually a really good one, I didn't know about `transpose`. Thanks for the tip!